zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookSocket.java

```diff
@@ -42,7 +42,7 @@ public String getHeader(String key) {
     return String.valueOf(headers.get(key));
   }
 
-  public void send(String serializeMessage) throws IOException {
+  public synchronized void send(String serializeMessage) throws IOException {
```

Comment:
synchronized method is not a proper solution for it. How about making it an asynchronous function?